### PR TITLE
Add Carthage support

### PR DESCRIPTION
**Pull request by klaaspieter**
Support Carthage by adding a framework scheme and target.

I'm aware there are two PRs (#53 and #55) open for this but they appear to be dead and have merge conflicts.

### CODE REVIEW

File: FBShimmering.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -232,15 +272,80 @@
...
+				MTL_ENABLE_DEBUG_INFO = NO;
+				PRODUCT_BUNDLE_IDENTIFIER = com.facebook.Shimmer;
+				PRODUCT_NAME = "$(TARGET_NAME)";
+				SKIP_INSTALL = YES;
+				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
+				SWIFT_VERSION = 3.0;
+				TARGETED_DEVICE_FAMILY = "1,2";
+				VERSIONING_SYSTEM = "apple-generic";
+				VERSION_INFO_PREFIX = "";
+			};
```
Comment (grp) [inline FBShimmering.xcodeproj/project.pbxproj]: Could you remove the unnecessary build settings set here? For example, no need for Metal and Swift related options.

File: Framework/Shimmer.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+#import <UIKit/UIKit.h>
+
+//! Project version number for Shimmer.
+FOUNDATION_EXPORT double ShimmerVersionNumber;
+
+//! Project version string for Shimmer.
+FOUNDATION_EXPORT const unsigned char ShimmerVersionString[];
+
```
Comment (grp) [inline Framework/Shimmer.h]: It doesn't look like you're defining these anywhere; remove them if they aren't set?

Comment (klaaspieter) [inline Framework/Shimmer.h]: This is what Xcode generates automatically. I always presumed there was a reason for it, but to be honest I don't know

Comment (grp) [inline Framework/Shimmer.h]: Ah, maybe this is paired with the versioning support file Xcode also generates. Either way, I don't see a good reason for it, so I'd be fine with disabling the versioning support in Xcode and removing it here.

File: FBShimmering.xcodeproj/project.pbxproj (hunk 1 above)
Comment (klaaspieter) [inline FBShimmering.xcodeproj/project.pbxproj]: 👍🏼 These are just the defaults generated by Xcode, but I'll remove the unnecessary ones.

**Comment by klaaspieter** [comment]
@grp I made the requested changes squashed them into a single commit.

**Comment by klaaspieter** [comment]
@grp anything else that needs to happen for this to get merged?

**Comment by klaaspieter** [comment]
@grp the other two PRs adding this can be closed to probably.

**Comment by bishalg** [comment]
@klaaspieter : there is still issue wth **Carthage** and the issue has not been fixed !

**Pull request merged**